### PR TITLE
NBSPlayer enhacements

### PR DESCRIPTION
**Pull request by TrentHouliston**
Some performance enhancements from the NBSPlayer in vision branches

### CODE REVIEW

File: src/server/nbs/nbs_nuclear_playback.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export class NbsNUClearPlayback extends stream.Writable {
   /** Convenience method for directly streaming a file to the network. */
   static fromFile(filename: string, nuclearnetClient: NUClearNetClient) {
     const playback = NbsNUClearPlayback.of(nuclearnetClient)
-    const rawStream = fs.createReadStream(filename)
+    const rawStream = fs.createReadStream(filename, { highWaterMark: 1024 * 1024 * 32 })
```
Comment (BrendanAnnable) [inline src/server/nbs/nbs_nuclear_playback.ts]: Add a comment to explain magic number?

File: src/server/nbs/nbs_frame_chunker.ts [hunk 2]
```diff
@@ -53,25 +52,21 @@ export class NbsFrameChunker extends stream.Transform {
...
-          }
+    if (headerIndex >= 0 && buffer.length > headerIndex + PACKET_SIZE_SIZE) {
+
+      // Read the size of the next packet
+      const sizeStart = headerIndex + NBS_HEADER.byteLength
+      const sizeEnd = sizeStart + PACKET_SIZE_SIZE
+      const packetSize = buffer.slice(sizeStart, sizeEnd).readUInt32LE(0)
+
+      // We have enough data
+      if (sizeEnd + packetSize <= buffer.length) {
```
Comment (BrendanAnnable) [inline src/server/nbs/nbs_frame_chunker.ts]: If I comment out this conditional the tests still pass, can we add a test for this?

File: package.json [hunk 3]
```diff
@@ -99,7 +98,6 @@
     "babel-preset-env": "^1.6.1",
     "binary": "^0.3.0",
     "bindings": "^1.3.0",
-    "buffers": "^0.1.1",
```
Comment (BrendanAnnable) [inline package.json]: 🎉

File: src/server/nbs/nbs_nuclear_playback.ts (hunk 1 above)
Comment (TrentHouliston) [reply src/server/nbs/nbs_nuclear_playback.ts]: Unfortunately, this magic number is literally magic. I don't know or remember where it came from :P

File: src/server/nbs/nbs_frame_chunker.ts (hunk 2 above)
Comment (TrentHouliston) [reply src/server/nbs/nbs_frame_chunker.ts]: Done

**Pull request merged**